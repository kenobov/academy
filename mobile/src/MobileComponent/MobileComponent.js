import React from 'react';
import {ApiServiceProvider} from "../context/ApiServiceContext";
import ApiService from "../services/ApiService";

import MobileTableComponent from "../MobileTableComponent/MobileTableComponent";
import MobileFilterComponent from "../MobileFilterComponent/MobileFilterComponent";
import {Button} from "reactstrap";
import inModal from "../hoc/inModal";
import MobileFormComponent from "../MobileFormComponent/MobileFormComponent";

class MobileComponent extends React.PureComponent {

    render() {

        const ModalButton = () => {
            return <Button color={"success"} size={"sm"}>
                Добавить клиента
            </Button>
        }
        const ModalOpener = inModal({title:'Редактирование клиента'})(ModalButton)(MobileFormComponent);

        return (
            <ApiServiceProvider value={new ApiService}>
                <div className={"card-body d-flex justify-content-between"}>
                    <MobileFilterComponent />
                    <ModalOpener initClient={initClient}/>
                </div>
                <MobileTableComponent />
            </ApiServiceProvider>
        )
    }

}

export default MobileComponent;

const initClient = {
    id: 0,
    surname: '',
    name: '',
    secondname: '',
    balance: 0,
    status: 'blocked',
}