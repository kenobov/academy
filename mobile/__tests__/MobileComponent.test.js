import React from 'react';
import renderer from 'react-test-renderer';

import MobileComponent from "../src/MobileComponent/MobileComponent";

test('Проверка фильтрации клиентов', () => {

    const component = renderer.create(
        <MobileComponent />
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const filterActiveClientsButton = component.root.find(
        el => el.props.id === 'filterActiveClients'
    );
    filterActiveClientsButton.props.onClick();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const filterBlockedClientsButton = component.root.find(
        el => el.props.id === 'filterBlockedClients'
    );
    filterBlockedClientsButton.props.onClick();

    tree = component.toJSON();
    expect(component).toMatchSnapshot();

    const filterAllClientsButton = component.root.find(
        el => el.props.id === 'filterAllClients'
    );
    filterAllClientsButton.props.onClick();

    tree = component.toJSON();
    expect(component).toMatchSnapshot();


});