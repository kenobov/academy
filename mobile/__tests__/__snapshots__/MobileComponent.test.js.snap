// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка фильтрации клиентов 1`] = `
Array [
  <div
    className="card-body d-flex justify-content-between"
  >
    <div
      className="btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        id="filterAllClients"
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterActiveClients"
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterBlockedClients"
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
    <a
      onClick={[Function]}
    >
      <button
        aria-label={null}
        className="btn btn-success btn-sm"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
    </a>
  </div>,
  <div
    className="Loader"
  >
    <div
      className="LoaderContent"
    >
      <div
        className="mt-5 mb-5 spinner-border text-primary"
        role="status"
      >
        <span
          className="visually-hidden"
        >
          Loading...
        </span>
      </div>
      Подождите, идет загрузка
    </div>
  </div>,
]
`;

exports[`Проверка фильтрации клиентов 2`] = `
Array [
  <div
    className="card-body d-flex justify-content-between"
  >
    <div
      className="btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterAllClients"
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        id="filterActiveClients"
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterBlockedClients"
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
    <a
      onClick={[Function]}
    >
      <button
        aria-label={null}
        className="btn btn-success btn-sm"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
    </a>
  </div>,
  <div
    className="Loader"
  >
    <div
      className="LoaderContent"
    >
      <div
        className="mt-5 mb-5 spinner-border text-primary"
        role="status"
      >
        <span
          className="visually-hidden"
        >
          Loading...
        </span>
      </div>
      Подождите, идет загрузка
    </div>
  </div>,
]
`;

exports[`Проверка фильтрации клиентов 3`] = `
Array [
  <div
    className="card-body d-flex justify-content-between"
  >
    <div
      className="btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterAllClients"
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterActiveClients"
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        id="filterBlockedClients"
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
    <a
      onClick={[Function]}
    >
      <button
        aria-label={null}
        className="btn btn-success btn-sm"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
    </a>
  </div>,
  <div
    className="Loader"
  >
    <div
      className="LoaderContent"
    >
      <div
        className="mt-5 mb-5 spinner-border text-primary"
        role="status"
      >
        <span
          className="visually-hidden"
        >
          Loading...
        </span>
      </div>
      Подождите, идет загрузка
    </div>
  </div>,
]
`;

exports[`Проверка фильтрации клиентов 4`] = `
Array [
  <div
    className="card-body d-flex justify-content-between"
  >
    <div
      className="btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        id="filterAllClients"
        onClick={[Function]}
        type="button"
      >
        Все
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterActiveClients"
        onClick={[Function]}
        type="button"
      >
        Активные
      </button>
      <button
        aria-label={null}
        className="btn btn-secondary"
        id="filterBlockedClients"
        onClick={[Function]}
        type="button"
      >
        Заблокированные
      </button>
    </div>
    <a
      onClick={[Function]}
    >
      <button
        aria-label={null}
        className="btn btn-success btn-sm"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
    </a>
  </div>,
  <div
    className="Loader"
  >
    <div
      className="LoaderContent"
    >
      <div
        className="mt-5 mb-5 spinner-border text-primary"
        role="status"
      >
        <span
          className="visually-hidden"
        >
          Loading...
        </span>
      </div>
      Подождите, идет загрузка
    </div>
  </div>,
]
`;
